src/toxcore/dht/server.rs

```diff
@@ -338,6 +356,7 @@ impl Server {
                 for node in &payload.nodes {
                     state.kbucket.try_add(node);
                 }
+                client.last_resp_time = Instant::now();
```

Comment:
Update only on ping response by design.

Reply:
https://zetok.github.io/tox-spec/#dht-operation

Nodes are removed after 122 seconds of no response. Nodes are added to the lists after a valid ping response or send node packet is received from them. If the node is already present in the list it is updated if the IP address changed.

I think above statements is for both PingResponse and NodesResponse

Reply:
Don't think, find corresponding code in C tox impl and compare it to docs.